src/resolvers/default.ts

```diff
@@ -1,34 +1,35 @@
 import { Resolver } from 'dns/promises'
 import { RecordType } from '../index.js'
-import { getTypes } from '../utils/get-types.js'
+import { convertType, getTypes } from '../utils/get-types.js'
 import { toDNSResponse } from '../utils/to-dns-response.js'
 import type { DNSResolver } from './index.js'
-import type { Answer } from '../index.js'
+import type { Answer, RecordTypeLabel } from '../index.js'
 
 const nodeResolver: DNSResolver = async (fqdn, options = {}) => {
   const resolver = new Resolver()
   const listener = (): void => {
     resolver.cancel()
   }
-  const types = getTypes(options.types)
+  const types = getTypes(options.types, options.useRecordTypeValue)
 
   try {
     options.signal?.addEventListener('abort', listener)
 
     const answers = await Promise.all(types.map(async type => {
-      if (type === RecordType.A) {
+      const valueType = convertType(type, options.useRecordTypeValue)
```

Comment:
will fix this... valueType should always be `RecordType` but here, it can be either `RecordType | RecordTypeLabel`